### PR TITLE
[WIP] GraphQL batch react client

### PR DESCRIPTION
**Pull request by ezr-ondrej**
[WIP] Description is a WIP.

### CODE REVIEW

File: app/views/dashboard/index.html.erb [hunk 1]
```diff
@@ -2,6 +2,9 @@
 <% title _('Overview') %>
 <%= title_actions dashboard_actions %>
 
+      <div id="graphql-test"></div>
+      <%= mount_react_component('GraphQLTest', '#graphql-test') %>
```
Comment (ezr-ondrej) [inline app/views/dashboard/index.html.erb:6]: [TODO] delete - Just to see it right away. Should go away.

**Comment by ezr-ondrej** [comment]
For the record, @ohadlevy raised a concern if we should batch everything, because some data fetching is taking a lot of time. So we will need eighter introduce an option for separate some requests, or use old API calls for such cases.

File: webpack/assets/javascripts/react_app/components/GraphQLTest/index.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import gql from 'graphql-tag';
+
+import { withGraphQLData } from '../../common/graphql';
+
+const GET_MODELS = gql`
```
Comment (xprazak2) [inline webpack/assets/javascripts/react_app/components/GraphQLTest/index.js:6]: We might consider using [webpack loader](https://medium.com/modus-create-front-end-development/load-your-graphql-documents-with-webpack-apollo-c1e166404cba) in the future, probably ok to keep the queries together with components for now.

File: webpack/assets/javascripts/react_app/common/graphql.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  cache: new InMemoryCache(),
+});
+
+const graphQLWrapperFactory = () => WrappedComponent => props => (
+  <ApolloProvider client={client}>
+    <WrappedComponent {...props} />
+  </ApolloProvider>
+);
+
+const withGraphQLData = (Component, query) => props => (
```
Comment (xprazak2) [inline webpack/assets/javascripts/react_app/common/graphql.js:32]: :star:

**Comment by xprazak2** [comment]
needs a rebase

**Pull request closed without merging**